### PR TITLE
Theme update and index page styling

### PR DESCRIPTION
**Pull request by thijsmie**

### CODE REVIEW

File: pelicanconf.py [hunk 1]
```diff
@@ -48,7 +49,7 @@
 
 #### SPONSORS ####
 SPONSORS = (('Eclipse Foundation', 'https://www.eclipse.org', 'images/eclipse-foundation.svg'),
-            ('ADLINK Technology', 'https://www.adlinktech.com', 'images/adlink.png'),)
+            ('ADLINK Technology', 'https://www.adlinktech.com', 'images/company_logo.svg'),)
```
Comment (k0ekk0ek) [inline pelicanconf.py:52]: Better name this `adlink.svg` since other companies might start contributing as well. e.g. APEX.AI might want it's logo here.

**Review by k0ekk0ek — CHANGES_REQUESTED** [review]
Thanks @thijsmie! Mostly LGTM, just the name thingy and I'll be happy to merge :+1:

File: pelicanconf.py [hunk 2]
```diff
@@ -61,8 +62,15 @@ def page_with_slug_index(pages):
...
             return page
 
+def page_with_slug(pages, slug):
+    for page in pages:
+        if page.slug == slug:
+            return page
+
 JINJA_FILTERS = {
-    'page_with_slug_index': page_with_slug_index, }
+    'page_with_slug_index': page_with_slug_index,
```
Comment (k0ekk0ek) [inline pelicanconf.py:71]: We can probably remove the `page_with_slug_index` entirely if we have a more generic one(?)

**Review by k0ekk0ek — CHANGES_REQUESTED** [review]

**Comment by k0ekk0ek** [comment]
I think I like the poetry thing as well, perhaps we should make that mandatory? And is there a nice pythonic way of downloading the resources?

**Comment by thijsmie** [comment]
Making poetry mandatory is 'hard' in the sense that it is just a package managing tool and there is nothing stopping people from just running `pip install pelican`. However, I do think it is a good idea to recommend in the Readme to use `poetry install` instead.

**Comment by k0ekk0ek** [comment]
Looks great @thijsmie, thanks!

**Pull request merged**